adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java

```diff
@@ -307,4 +308,13 @@ private Integer getIntegerQueryParameter(final String parameterName) {
                 .orElse(null);
     }
 
+    /**
+     * Get the QoS level as set in the request by the device.
+     *
+     * @return The QoS level requested by the device.
+     */
+    public QoS getRequestedQos() {
```

Comment:
IMHO we should add this method to `ExecutionContext` because it is a general requirement for all adapters/protocols.

Reply:
Are you working on this? I can't find a corresponding change in your latest push?

Reply:
Yes, I do! I will push a separate commit containing the QoS changes soon.

Reply:
There you go, see: 9bb5266

Let me know what you think about the two QoS classes and the `NOT_APPLICABLE` and the `UNKNOWN` levels

Reply:
Since this is the only conversation left unresolved: is there something I missed or are you still reviewing?